torch/csrc/jit/tensorexpr/external_functions.cpp

```diff
@@ -119,6 +72,110 @@ at::Tensor from_blob_quantized(
   return qx;
 }
 
+std::vector<at::Tensor> constructTensors(
+    int64_t bufs_num,
+    void** buf_data,
+    int64_t* buf_ranks,
+    int64_t* buf_dims,
+    int64_t* buf_strides,
+    int8_t* buf_dtypes,
+    c10::optional<std::vector<std::pair<size_t, QIData>>> qdataArg) {
+  std::vector<void*> buf_data_vec;
+  std::vector<std::vector<int64_t>> buf_dims_vec;
+  std::vector<std::vector<int64_t>> buf_strides_vec;
+  std::vector<c10::ScalarType> buf_dtypes_vec;
+  int64_t buf_dims_idx = 0;
+  int64_t buf_strides_idx = 0;
+  for (const auto i : c10::irange(bufs_num)) {
+    buf_data_vec.push_back(buf_data[i]);
+    buf_dims_vec.emplace_back();
+    buf_strides_vec.emplace_back();
+    for (const auto dim : c10::irange(buf_ranks[i])) {
+      (void)dim;
+      buf_dims_vec[i].push_back(buf_dims[buf_dims_idx++]);
+      buf_strides_vec[i].push_back(buf_strides[buf_strides_idx++]);
+    }
+    buf_dtypes_vec.push_back(static_cast<c10::ScalarType>(buf_dtypes[i]));
+  }
+
+  std::vector<at::Tensor> tensors;
+  if (!qdataArg.has_value()) {
+    for (const auto i : c10::irange(buf_data_vec.size())) {
+      auto options = at::TensorOptions()
+                         // NOLINTNEXTLINE
+                         .dtype(buf_dtypes_vec[i])
+                         .layout(at::kStrided)
+                         .device(at::kCPU) // TODO: support GPUs too
+                         .memory_format(deduce_memory_format(
+                             // NOLINTNEXTLINE
+                             buf_strides_vec[i],
+                             // NOLINTNEXTLINE
+                             buf_dims_vec[i]))
+                         .requires_grad(false);
+      auto tensor = at::from_blob(
+          // NOLINTNEXTLINE
+          buf_data_vec[i],
+          buf_dims_vec[i],
+          buf_strides_vec[i],
+          options);
+      tensors.emplace_back(tensor);
+    }
+  } else {
+    // handle quantized
+    std::vector<c10::optional<QIData>> qdata(bufs_num, c10::nullopt);
+    for (const auto& qd : *qdataArg) {
+      qdata[qd.first] = qd.second;
+    }
+    for (const auto i : c10::irange(buf_data_vec.size())) {
+      auto options = at::TensorOptions()
+                         // NOLINTNEXTLINE
+                         .dtype(buf_dtypes_vec[i])
+                         .layout(at::kStrided)
+                         .device(at::kCPU) // TODO: support GPUs too
+                         .memory_format(deduce_memory_format(
+                             // NOLINTNEXTLINE
+                             buf_strides_vec[i],
+                             // NOLINTNEXTLINE
+                             buf_dims_vec[i]))
+                         .requires_grad(false);
+      if (auto qd = qdata[i]) {
```

Comment:
Can you separate out assign from the if for better readability?

Reply:
@kimishpatel Why does it make code less readable ? :) 
I thought the main point of introducing initialization inside the if conditions was to increase readability:
https://skebanga.github.io/if-with-initializer/

Reply:
I was mainly coming from "hey is this a typo?" and then I realized no it isnt. The argument in the link seems to be more from the perspective of more verbose code specifically due to the usage pattern it has. I am fine if you want to keep it this way. It is matter of personal preference.